src/ImageSharp/Color/Color.cs

```diff
@@ -124,7 +125,9 @@ public static Color FromPixel<TPixel>(TPixel pixel)
         {
             return new((L16)(object)pixel);
         }
-        else if (Unsafe.SizeOf<TPixel>() <= Unsafe.SizeOf<Rgba32>())
+
```

Comment:
While not critical, I would prefer an abstraction that can get rid of the special casing above this line - by moving it to `IPixel` / `PixelOperations` code.

Reply:
Me also. In this case I think we should ditch the specialisation and box anything greater than byte. `ColorFromPixel` is not area for high performance operations.

On `IPixel` I would, if we can, drop all the various `FromXX` specialisation there also and ensure a good optimised bulk operations for all `Rgba32` compatible pixel formats. That may not be possible though.

Reply:
Well remembered, but those are really short loops (max 256) in real terms. I'd hate to limit our flexibility and extensibility based on introducing a requirement there. 

Ideally, I want to introduce as much agnostity as possible in our APIs.

Reply:
Boxing is very ugly, even if done only `256x` typically, I would try to avoid it if there are alternatives.

Reply:
Maybe we just back Color with `Vector4`? it already takes up the same amount of space and means we would be able to use bulk operations.

Reply:
The reason we introduced the ability to define (higher than `Vector4`) arbitrary precison pixel types was this feature request: https://github.com/SixLabors/ImageSharp.Drawing/discussions/165

I like the fact that ImageSharp supports such use-cases.

Reply:
Then use `Vector4` as the default and  box anything bigger. People expected double precision are in for a rough time with the library since almost everything is converted to use single precision at some point.

Reply:
I've updated `Color` to use `Vector4` to avoid boxing. I've also removed the implicit casting for random pixel formats in favour of explicit methods.